interface/wx/pen.h

```diff
@@ -70,6 +70,31 @@ enum wxPenStyle
         /**< Last of the hatch styles (inclusive). */
 };
 
+/**
+    Possible values for pen quality.
+
+    Pen quality is currently only used in wxMSW, the other ports ignore it and
+    always use the same default pen quality.
+
+    In wxMSW the choice of quality affects whether "cosmetic" or "geometric"
+    native pens are used in the situations when both are usable. Notably, for
+    the dotted and dashed pens of width 1, high quality geometric pens are used
+    by default since wxWidgets 3.1.4, while previous versions used lower
+    quality but much faster cosmetic pens. If drawing performance is more
+    important than the exact appearance of the lines drawn using this pen, low
+    quality may be explicitly selected.
```

Comment:
> In case it's unexpected: I initially ran the tests without `--pen-style=dot` and then the lines are mostly black (same as 3.1.4), but with `--pen-quality` set to either `low` or `high` they then appear white. With the default quality there seems to be a single black pen created, while with `low` or `high` black is initially also used once but multiple pens after that are white. This was with XP.

This is just the test not bothering to set the pen at all without either pen width or pen style (or pen quality, now) option. We could, and probably should, make the colour the same as the default one, I guess.

As for "the"s, I was thinking about "those specific situations" and "these particularly-styled pens", but maybe it's a wrong thing to put the accent on here, I don't know. It surely flows a bit better without them, but the meaning seems slightly different to me. But maybe I'm just imagining things.

Reply:
> This is just the test not bothering to set the pen at all without either pen width or pen style (or pen quality, now) option. We could, and probably should, make the colour the same as the default one, I guess.

Thanks for explaining, yes it's minor but at least wouldn't have lead me to think this was somehow related to the library changes (as opposed to the test, I now understand it's the `setPen` changes at play).

> As for "the"s, I was thinking about "those specific situations" and "these particularly-styled pens", but maybe it's a wrong thing to put the accent on here, I don't know. It surely flows a bit better without them, but the meaning seems slightly different to me. But maybe I'm just imagining things.

Hmm maybe it could slightly change the meaning here, but I don't know if it does in a meaningful way. Rereading it over and over also doesn't help... Plus not a native speaker of course. Perhaps it's about flow indeed, it's always been somewhat noticeable when reading (and got me thinking .fr throws articles in front of nouns at random ;) ).

Ridiculous slowdowns btw. I have been comparing v3.1.4 performance of `--pen-style=dot --lines -N 10000` and `--lines -N 10000` and even `wxClientDC` is easily 100x slower (nothing like a 16x slowdown as mentioned in the ticket). Some crosshairs manage almost 500x, and horizontal dotted lines are much slower than vertical ones for some reason.
This was in a VM with XP however and I also didn't pay attention to whether the pointer obscured the window or not. Will try on an actual machine when I can and with something more modern OS-wise to see if the numbers are also that bad.